### PR TITLE
feat: nodejs application

### PR DESCRIPTION
**Pull request by glasnt**
Implements the nodejs application. 

With assistance from @pattishin ✨

### CODE REVIEW

File: app/nodejs/process.js [hunk 1]
```diff
@@ -20,15 +20,154 @@ process.js - take data in raw_data bucket, process,
...
+
+  let counter = 0;
+
+  const processedBucket = storage.bucket(config.PROCESSED_DATA_BUCKET);
+
+  const writeData = new Promise((resolve, reject) => {
+    Object.keys(aggregate).forEach(async function(rowKey) {
+      const facetData = aggregate[rowKey];
+      processedBucket
+          .file(`${rowKey}/promise_data.json`)
```
Comment (NimJay) [inline app/nodejs/process.js:148]: Should this be:
```suggestion
          .file(`${rowKey}/data.json`)
```
(to match the [Python version](https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/blob/main/app/python/process.py#L132C32-L132C41)?)

Comment (glasnt) [reply app/nodejs/process.js:148]: Ah yes, this was a debugging step for me (so I could compare data and promise_data to confirm they were the same)

File: app/nodejs/app.js [hunk 2]
```diff
@@ -19,26 +19,65 @@ app.js - application
...
+      `Has the job been run?`);
+    return 0, [];
+  }
+  const fragment = await dataFile.download();
+  const data = JSON.parse(fragment);
+  const squirrelCount = data._counter;
+  delete data._counter;
+
+  logger.info(`Retrieved data for ${squirrelCount} entities.`);
+  return [squirrelCount, Object.values(data)];
```
Comment (NimJay) [inline app/nodejs/app.js:59]: FYI - related pull-request: https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/pull/36/files

**Comment by glasnt** [comment]
Requires more complete unit tests, outside of my current wheelhouse (see python implementation for an example, should be infra-free (mocking where required))

**Comment by glasnt** [comment]
WIP application mocking in [df9e397](https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/pull/30/commits/df9e397ad67affe429ecef699a107290623dc58e)

https://jestjs.io/docs/ecmascript-modules may make it difficult. Tried using sinon and esmock to mock the `retrieveData` method, but was unable to get it working

**Comment by glasnt** [comment]
Infra tests currently failing pre-job execution. 
Service 503; logs: 

```
ApiError: No such object: processed-client-api-NNNN-pppppppp/Gray/Adult/Above Ground/data.json
```

Should return "No data available. Follow the tutorial instructions." (with link to neos tutorial)

File: app/nodejs/app.js [hunk 3]
```diff
@@ -19,26 +19,71 @@ app.js - application
...
+ * @param {string} location Location (Above Ground, Ground Level)
+ */
+async function retrieveData(fur, age, location) {
+  const dataFilename = [fur, age, location].join('/') + '/data.json';
+
+  const dataFile = storage
+      .bucket(process.env.PROCESSED_DATA_BUCKET)
+      .file(dataFilename);
+
+  if (!dataFile.exists()) {
```
Comment (glasnt) [inline app/nodejs/app.js:46]: This isn't returning synchronously, and a promise is truthy, so the logic here doesn't fire if the file doesn't exist.

Comment (glasnt) [reply app/nodejs/app.js]: Fixed it (hopefully!)

**Comment by glasnt** [comment]
unit test fix up originally in https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/commit/9818de0e4ec2abdb771fbb37afb9ad38b74331f5, manually applied and co-author attribution added

**Comment by glasnt** [comment]
[df9e72b](https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/pull/30/commits/df9e72bba24b9b6c582c987f5c951d1227ffe689) introduces unique-tagged image for validation testing. Needs to be reverted before merging.

**Comment by glasnt** [comment]
Linting issues means the previous commit was erroneous. [9959e0d](https://github.com/GoogleCloudPlatform/terraform-cloud-client-api/pull/30/commits/9959e0d737ac6636c41596ffc12d825a3f3c1f98) should affect the same change.

File: infra/main.tf [hunk 4]
```diff
@@ -17,7 +17,8 @@
 locals {
   unique = "${var.deployment_name}-${random_id.default.hex}"
 
-  application_image = "us-docker.pkg.dev/hsa-public/containers/cloud-client-api/${lower(var.language)}:${var.image_version}"
+  temp_image_version = "${var.image_version}-pr30"
```
Comment (NimJay) [inline infra/main.tf:20]: **Question:**
For now, if we make Java/Python/JavaScript source code changes, we use `temp_image_version` to test those changes.
Before merging, we would revert to `var.image_version`.
Future: We'll think of a solution for this chicken-and-egg `temp_image_version` problem and remove `temp_image_version` entirely.
Are my assumptions correct?

If so, this sounds good to me. :)
Again, thanks for being so thorough.

**Review by NimJay — APPROVED** [review]
Looks good to me! Left two comments — nothing major.
Feel free to merge whenever.

File: infra/main.tf (hunk 4 above)
Comment (glasnt) [reply infra/main.tf:20]: I'll ping you internal docs on how this is intended to work

**Pull request merged**